### ISSUE TITLE
Errors with Vite + Vue3

### ISSUE THREAD

**Issue by changefox** [imanmalekian31/vue-bottom-navigation#17]
Love the look and feel of this but having issues when using it with Vite + Vue 3

```✘ [ERROR] Could not resolve "./components/CurvedBottomNavigation"

    node_modules/bottom-navigation-vue/src/install.js:1:35:
      1 │ ... CurvedBottomNavigation from "./components/CurvedBottomNavigation";
        ╵                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] Could not resolve "./components/GrowBottomNavigation"

    node_modules/bottom-navigation-vue/src/install.js:2:33:
      2 │ import GrowBottomNavigation from "./components/GrowBottomNavigation";
        ╵                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] Could not resolve "./components/SwipeBottomNavigation"

    node_modules/bottom-navigation-vue/src/install.js:3:34:
      3 │ ...rt SwipeBottomNavigation from "./components/SwipeBottomNavigation";
        ╵                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] Could not resolve "./components/RingBottomNavigation"

    node_modules/bottom-navigation-vue/src/install.js:4:33:
      4 │ import RingBottomNavigation from "./components/RingBottomNavigation";
        ╵                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] Could not resolve "./components/WindowsBottomNavigation"

    node_modules/bottom-navigation-vue/src/install.js:5:36:
      5 │ ...indowsBottomNavigation from "./components/WindowsBottomNavigation";
        ╵                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

1:32:15 pm [vite] error while updating dependencies:
Error: Build failed with 5 errors:
node_modules/bottom-navigation-vue/src/install.js:1:35: ERROR: Could not resolve "./components/CurvedBottomNavigation"
node_modules/bottom-navigation-vue/src/install.js:2:33: ERROR: Could not resolve "./components/GrowBottomNavigation"
node_modules/bottom-navigation-vue/src/install.js:3:34: ERROR: Could not resolve "./components/SwipeBottomNavigation"
node_modules/bottom-navigation-vue/src/install.js:4:33: ERROR: Could not resolve "./components/RingBottomNavigation"
node_modules/bottom-navigation-vue/src/install.js:5:36: ERROR: Could not resolve "./components/WindowsBottomNavigation"
    at failureErrorWithLog (/Users/user/Development/app/node_modules/esbuild/lib/main.js:1603:15)
    at /Users/user/Development/app/node_modules/esbuild/lib/main.js:1249:28
    at runOnEndCallbacks (/Users/user/Development/app/node_modules/esbuild/lib/main.js:1034:63)
    at buildResponseToResult (/Users/user/Development/app/node_modules/esbuild/lib/main.js:1247:7)
    at /Users/user/Development/app/node_modules/esbuild/lib/main.js:1356:14
    at /Users/user/Development/app/node_modules/esbuild/lib/main.js:666:9
    at handleIncomingPacket (/Users/user/Development/app/node_modules/esbuild/lib/main.js:763:9)
    at Socket.readFromStdout (/Users/user/Development/app/node_modules/esbuild/lib/main.js:632:7)
    at Socket.emit (events.js:375:28)
    at addChunk (internal/streams/readable.js:290:12)
Vite Error, /node_modules/.vite/deps/bottom-navigation-vue.js?v=3a7d9c9a optimized info should be defined
Vite Error, /node_modules/.vite/deps/bottom-navigation-vue.js?v=3a7d9c9a optimized info should be defined (x2)
Vite Error, /node_modules/.vite/deps/bottom-navigation-vue.js?v=3a7d9c9a optimized info should be defined (x3)
```

main.js
```import { createApp } from "vue";

import router from "@/router";
import { plugin, defaultConfig } from "@formkit/vue";
import { generateClasses } from "@formkit/tailwindcss";
import { createPinia } from "pinia";
import App from "./App.vue";
import theme from "./theme.js";
import "./index.css";
import VueCountdown from "@chenfengyuan/vue-countdown";
import { GrowBottomNavigation } from "bottom-navigation-vue";

createApp(App)
  .component(VueCountdown.name, VueCountdown)
  .use(createPinia())
  .use(GrowBottomNavigation)
  .use(
    plugin,
    defaultConfig({
      config: {
        classes: generateClasses(theme),
      },
    })
  )
  .use(router)
  .mount("#app");
```

MobileNavigation.vue
```
<template>
  <GrowBottomNavigation :options="options" v-model="selected" />
</template>

<script>
import { GrowBottomNavigation } from "bottom-navigation-vue";

export default {
  components: { GrowBottomNavigation },
  data: () => ({
    selected: 1,
    options: [
      { id: 1, icon: "fa-solid fa-home", title: "Home", color: "#5639af" },
      {
        id: 2,
        icon: "fa-solid fa-magnifying-glass",
        title: "Search",
        color: "#ac4793",
      },
      { id: 3, icon: "fa-solid fa-heart", title: "Likes", color: "#e2a93a" },
      { id: 4, icon: "fa-solid fa-gear", title: "Settings", color: "#4493a7" },
    ],
  }),
};
</script>
```

I've tried with/without the main.js import and with/without the component import with the same issue. Not sure if I'm doing something stupid or not

**Comment by changefox** [comment]
I'd also prefer to use the Composition API if possible (to keep my code consistent).

**Comment by changefox** [comment]
Did some exploration and found;
 - Imports should include the .vue file

```
import CurvedBottomNavigation from "./components/CurvedBottomNavigation.vue";
import GrowBottomNavigation from "./components/GrowBottomNavigation.vue";
import SwipeBottomNavigation from "./components/SwipeBottomNavigation.vue";
import RingBottomNavigation from "./components/RingBottomNavigation.vue";
import WindowsBottomNavigation from "./components/WindowsBottomNavigation.vue";

export {
  CurvedBottomNavigation,
  GrowBottomNavigation,
  SwipeBottomNavigation,
  RingBottomNavigation,
  WindowsBottomNavigation
};
```
 - this.$set is not used in Vue 3 in GrowBottomNavigation.vue
```
  created() {
    this.localOptions = this.options.slice();

    const index = this.localOptions.findIndex(
      (item) =>
        item.id == this.value ||
        (item.path || {}).name == (this.$route || {}).name
    );

    if (index > -1) {
      this.currSelected = index;
      this.prevSelected = index;

      this.localOptions[index].selected = true;

      // this.$set(this.localOptions, index, {
      //   ...this.localOptions[index],
      //   selected: true
      // });
    }
  },
```


Hopefully this helps as you make this package work with Vue 3 in the future. Sorry I did think that this was Vue 3 compliant (not sure why)

**Comment by imanmalekian31** [comment]
Thank you for searching for it.
I will work on it.

**Comment by imanmalekian31** [comment]
I fix this problem in the new version (`2.5.0`).
I hope you enjoy that.

**Comment by changefox** [comment]
Thank you for the update. works as expected now :)
